server/src/api/modules/groups/services/ChangeMemberRoleService.ts

```diff
@@ -16,50 +19,75 @@ type ChangeMemberRoleService = z.infer<typeof inputSchema>;
 async function changeMemberRole(payload: ChangeMemberRoleService): Promise<MembershipWithPlayer> {
   const params = inputSchema.parse(payload);
 
-  const membership = await prisma.membership.findFirst({
-    where: {
-      groupId: params.id,
-      player: { username: standardize(params.username) }
-    }
-  });
-
-  if (!membership) {
-    const group = await prisma.group.findFirst({
-      where: { id: params.id }
-    });
+  const result = await prisma
+    .$transaction(async transaction => {
```

Comment:
I don't think this should all be inside a transaction, only:

- transaction.membership.update(..)
- transaction.memberActivity.create(...)